src/runtime/components/data/Table.vue

```diff
@@ -349,11 +353,12 @@ export default defineComponent({
       return get(row, rowKey, defaultValue)
     }
 
-    function toggleOpened (index: number) {
+    function toggleOpened ({ index, row }: { index: number, row: Row }) {
       if (openedRows.value.includes(index)) {
         openedRows.value = openedRows.value.filter((i) => i !== index)
       } else {
         openedRows.value.push(index)
+        emit('update:expand', { expandedRows: openedRows.value, row })
```

Comment:
Shouldn't this be emitted in both case?

Reply:
Should it? I think it could be either emitted in both cases and the information whether it was an "expand" or "collapse" could be added as a function parameter to the user or there could be another emit like `update:collapse`. What do you think?

Reply:
Since you send the `openedRows` in the event, we should emit each time the `openedRows.value` changes I guess.